hardhat.config.ts

```diff
@@ -52,16 +52,16 @@ const config: HardhatUserConfig = {
     },
   },
   solidity: {
-    version: '0.8.7',
+    version: '0.8.12',
     settings: {
       optimizer: {
         enabled: true,
         runs: 200,
         details: {
-          yul: false,
```

Comment:
What do yul and viaIR flags do? Maybe some comments or we link to the reference for this at top of the file with just one link.

Reply:
via-ir is Solidity compilers newer way to compile to bytecode. Basically instead of optimizing the solidity=>opcode it optimizes solidity=>yul=>bytecode, but it opens up alot more optimizations. I have had great results on gas and contract size with this. Here are the docs. Even though i did see when i looked that up, it was still considered experimental 8.12. it was considered stable in 8.13 may want to bump up compiler one more version
https://docs.soliditylang.org/en/v0.8.13/ir-breaking-changes.html?highlight=via-ir